packages/animations/lib/src/page_transition_switcher.dart

```diff
@@ -43,6 +58,16 @@ class _ChildEntry {
   }
 }
 
+/// Signature for builders used to generate custom layouts for
+/// [PageTransitionSwitcher].
+///
+/// The builder should return a widget which contains the given children, laid
+/// out as desired. It must not return null. The builder should be able to
+/// handle an empty list of `_activeEntries`.
+typedef PageTransitionSwitcherLayoutBuilder = Widget Function(
+  List<ChildEntry> _activeEntries,
```

Comment:
Why do we hand the full `ChildEntry` to the layoutBuilder? It seems that handing them just the `transition` Widgets from the ChildEntries would be enough. The layout builder shouldn't have access to the animation controllers in `ChildEntry`. Those shouldn't be messed with. 

Similarly to `AnimatedSwitcher`'s layoutBuilder it probably also makes sense to hand over the currentEntry and all other entries in separate arguments.

Reply:
In other words, `_ChildEntry` should stay private.

Reply:
In other words, `_ChildEntry` should stay private.